enhancements/microshift/microshift-default-cni.md

```diff
@@ -0,0 +1,290 @@
+requirements of the edge environment, an optimized version of OVN Kubernetes
+is proposed.
+
+This option aligns better with Red Hat's strategy on networking and avoids
+the need to maintain bridge-cni which is deprecated in podman 4.
+
+Using OVN-kubernetes is only possible thanks to the following optimizations:
+* Single node operation
+* Communication via unix sockets, avoiding the need of TLS & Certificate creation
+  at boot time
+* No leader election, as there is no high availability or multi-node operation.
+* OpenvSwitch and ovsdb-server RAM optimization (CPUAffinity=0 and --no-mlockall)
+* Workload partitioning, to limit the cpusets where specific ovn-kubernetes workloads
+  can run.
+* New ovn-kubernetes with reduced footprint, oriented to single node operation.
+
+The optimizations reduce the total footprint impact of OVN-Kubernetes + OpenvSwitch
+down to:
+* 100MB of RAM (from +600MB)
+* 366MB of Disk (from 1GB)
+* 0.011CPU cores (based on an i7-7600 platform or a Xeon 4210 VM with 1 VCPU)
+
+Remaining work items in ovn-kubernetes:
+* Handling node IP changes
+* Improve cache allocation in libovsdb (this accounts for 16MB on northd and southdb
+  database clients)
+
+### Workflow Description
+
+#### Deploying
+
+MicroShift should setup the CNI during startup, and any specific configurations
+to cri-o or the system should be handled by the RPM install of `microshift-networking`.
+
+#### Upgrading
+
+Upgrades to the OVN database should be handled by the ovn-dbchecker during
+the ovn startup process.
+
+#### Configuring
+
+MicroShift reads its configuration file once on startup and feeds any
+necessary network details (like Pod or Service Cluster IP CIDRs down
+to the CNI)
+
+#### Deploying Applications
+
+Applications are deployed on MicroShift using standard Kubernetes
+resources such as Deployments, Services or NetworkPolicies. 
+They can be deployed at runtime via API calls, or embedded in the system
+image with the manifests loaded from the filesystem by MicroShift on startup.
+By using the `IfNotPresent` pull policy and adding images to the CRI-O cache
+in the system image, it is possible to build a device that can boot and
+launch the application without network access.
+
+### API Extensions
+
+The MicroShift default CNI does not add any APIs to OpenShift.
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+Implementing OVNKubernetes without the ClusterNetworkOperator and under
+an optimized topology requires a different set of deployment manifests,
+those manifests are embedded into MicroShift and the combination of
+MicroShift plus CNI should be subject to the same amount of testing
+OVN Kubernetes is, excluding not supported features like multi-node.
+
+Footprint optimizations applied today could stop being effective in the
+long term. To mitigate this risk continous performance/footprint analysis
+must be implemented in MicroShift.
```

Comment:
How is this going to be achieved? Should it be part of the test plan?

Reply:
Yes that's the idea. I will add it to the test plan, good point.